string.go

```diff
@@ -7,6 +7,12 @@ type String string
 
 func (s String) Error() string { return string(s) }
 
+// Is returnes true if the target is the same underlying
+// String value (compared as a string)
+func (s String) Is(target error) bool {
+	return s == target
+}
```

Comment:
I agree `errorString.Unwrap` is needed (good catch), but I don't think this is.  I think the stdlib already takes care of this.  Does the test you introduced fail without this?